enterprise/core-edge/src/main/java/org/neo4j/coreedge/server/core/CoreServerStartupProcess.java

```diff
@@ -85,10 +83,12 @@ public void start() throws Throwable
             {
                 caughtUp.get( joinCatchupTimeout, MILLISECONDS );
             }
-            catch ( InterruptedException | ExecutionException | TimeoutException e )
+            catch ( Throwable e )
             {
-                throw new RuntimeException( format( "Server failed to join cluster within catchup time limit [%d ms]",
-                        joinCatchupTimeout ), e );
+                String message =
+                        format( "Server failed to join cluster within catchup time limit [%d ms]", joinCatchupTimeout );
+                log.error( message, e );
```

Comment:
Isn't the issue that this simply is only printed to neo4j.log and it would be good if all lifecycle issues were also reported in debug.log? We should likely ask operability to fix that in such case.